conseil-common/src/main/scala/tech/cryptonomic/conseil/common/metadata/MetadataService.scala

```diff
@@ -17,22 +16,24 @@ class MetadataService(
     config: PlatformsConfiguration,
     transformation: UnitTransformation,
     cacheConfiguration: AttributeValuesCacheConfiguration,
-    platformDiscoveryOperations: PlatformDiscoveryOperations
+    platformDiscoveryOperations: Map[String, PlatformDiscoveryOperations]
 )(implicit apiExecutionContext: ExecutionContext) {
 
-  private val platforms = transformation.overridePlatforms(ConfigUtil.getPlatforms(config))
+  private val platforms = transformation.overridePlatforms(config.getPlatforms)
 
   private val networks = platforms.map { platform =>
-    platform.path -> transformation.overrideNetworks(platform.path, ConfigUtil.getNetworks(config, platform.name))
+    platform.path -> transformation.overrideNetworks(platform.path, config.getNetworks(platform.name))
   }.toMap
 
   private val entities = {
-    val allEntities = Await.result(platformDiscoveryOperations.getEntities, 2 second)
+    val allEntities = platforms
```

Comment:
Would be `traverse` the platforms and then `Await` for the whole list more clear?

Reply:
```
val futureEntities = platforms.map { platform =>
   platformDiscoveryOperations(platform.name).getEntities.map(v => platform.name -> v)
}
val allEntities = Await.result(Future.sequence(futureEntities), 5 second).toMap
```

That's the idea I managed to came up with. Would that work better for you or maybe you have better one?

Reply:
yeah, it's fine.

FYI `traverse` is exactly map+sequence, and it's most often what's actually needed

```scala
val futureEntities = Future.traverse(platforms) { platform =>
  platformDiscoveryOperations(platform.name)
    .getEntities.map(v => platform.name -> v)
}
val allEnties = Await.result(futureEntities, 5.seconds).toMap
```

Reply:
Awesome. I've changed that.